setup.cfg

```diff
@@ -77,12 +77,6 @@ docs =
 	sphinx-inline-tabs
 	sphinxcontrib-towncrier
 
-ssl =
-	wincertstore; sys_platform=='win32'
-
-certs =
-	certifi
```

Comment:
That is probably a really bad idea for certain kinds of Windows users, most macOS users, and Debian/Ubuntu users.

* https://github.com/psf/requests/issues/2966
* https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=960869

Reply:
previously setuptools only fell back on certifi if every system cert file was missing:

```
    return (
        get_win_certfile()
        or next(extant_cert_paths, None)
        or _certifi_where()
    )
```

so I think this is correct?

Reply:
@tiran as far as I can tell I can't install certifi without installing ca-certificates

Reply:
https://gist.github.com/tiran/2dec9e03c6f901814f6d1e8dad09528e#ssl-module-cannot-verify-connections

Reply:
Debian-based distros don't come with ca-certificates in their minimal package list because Debian-based distros use plain HTTP + GPG signatures instead of HTTPS to download and verify packages. Starting with Ubuntu >= 21.04 and current Debian Testing, the Python package has a **weak** dependency on ca-certificates.

Reply:
Yeah so as far as I can tell dropping this certifi dep doesn't make things worse for Debian/Ubuntu

Reply:
The same applies for Windows. ``ssl.enum_certificates`` no longer works in some cases because Windows retrieves trust anchors on demand.

Reply:
It seems windows `urllib.request.urlopen(..., context=ssl.create_default_context(capath=None))` works correctly too:

```
>>> import ssl
>>> urllib.request.urlopen("https://google.com", context=ssl.create_default_context(capath=None))
<http.client.HTTPResponse object at 0x0000011183D54130>
>>> urllib.request.urlopen("https://wrong.host.badssl.com/", context=ssl.create_default_context(capath=None))
Traceback (most recent call last):
  File "C:\Python39\lib\urllib\request.py", line 1346, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "C:\Python39\lib\http\client.py", line 1253, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "C:\Python39\lib\http\client.py", line 1299, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "C:\Python39\lib\http\client.py", line 1248, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "C:\Python39\lib\http\client.py", line 1008, in _send_output
    self.send(msg)
  File "C:\Python39\lib\http\client.py", line 948, in send
    self.connect()
  File "C:\Python39\lib\http\client.py", line 1422, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "C:\Python39\lib\ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "C:\Python39\lib\ssl.py", line 1040, in _create
    self.do_handshake()
  File "C:\Python39\lib\ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'wrong.host.badssl.com'. (_ssl.c:1129)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python39\lib\urllib\request.py", line 214, in urlopen
    return opener.open(url, data, timeout)
  File "C:\Python39\lib\urllib\request.py", line 517, in open
    response = self._open(req, data)
  File "C:\Python39\lib\urllib\request.py", line 534, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "C:\Python39\lib\urllib\request.py", line 494, in _call_chain
    result = func(*args)
  File "C:\Python39\lib\urllib\request.py", line 1389, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "C:\Python39\lib\urllib\request.py", line 1349, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'wrong.host.badssl.com'. (_ssl.c:1129)>
>>>
```

Reply:
My windows installation of setuptools doesn't have certifi or wincertstore installed

Reply:
@tiran I'm confident certifi is never used here, barring intentional user intervention.

The setuptools.package_index module is only used when pip is not available and setup.py is invoked with a deprecated `setup_requires` kwarg. On Windows and macos ensurepip works correctly so this isn't an issue.

further certifi is *only* used when ca-certificates is not installed. because on debian python3-certifi depends on ca-certificates this code is unreachable.

installing certifi manually with pip would also render this code unreachable because installing pip means this code won't run either

installing certifi with easyinstall isn't possible because certifi isn't installed and so certificate verification will fail

To reach this code a user would have to manually copy over the certifi files into their site-packages or add the whl to their sys.path

Reply:
It should also be noted that `setup.py upload` (also deprecated) uses vanilla urllib.request.urlopen without any certificate shenanigans https://github.com/pypa/setuptools/blob/582465bb1e4ba2c2bb34b580427634d045a1f723/setuptools/_distutils/command/upload.py#L194-L195